src/PageContent/AssignVolunteers/Edit.js

```diff
@@ -0,0 +1,133 @@
+                    <i className="fas fa-circle ellipses" />
+                    <h5 className="location" id="bottom"><i className="fas fa-circle circle-marker" /><span className="spacing" />{this.state.receivingAgency}</h5>
+                </div>
+                <div className="form-container">
+                    <form onSubmit={this.handleConfirmClick.bind(this)}>
+
+                        <label className="label-component">Student Deliverers</label>
+
+                        <div className="form-parent">
+
+                            <div className="form-child">
+                                <label className="label-component details">Student 1</label><br />
+                                <input name="name" type="text" className="form-input" onChange={this.handleNameOne.bind(this)} placeholder={this.state.s1name} required/><br />
+                                <label className="label-component details">Phone</label><br />
+                                <input name="phone" type="tel" className="form-input" onChange={this.handlePhoneOne.bind(this)} placeholder={this.state.s1phone} required/><br />
+                                <label className="label-component details">Email</label><br />
+                                <input type="email" className="form-input" onChange={this.handleEmailOne.bind(this)} placeholder={this.state.s1email} required/>
+                            </div>
+                            <div className="form-child">
+                                <label className="label-component details">Student 2</label><br />
+                                <input name="name" type="text" className="form-input" onChange={this.handleNameTwo.bind(this)} placeholder={this.state.s2name} required/><br />
+                                <label className="label-component details">Phone</label><br />
+                                <input name="phone" type="tel" className="form-input" onChange={this.handlePhoneTwo.bind(this)} placeholder={this.state.s2phone} required/><br />
+                                <label className="label-component details">Email</label><br />
+                                <input type="email" className="form-input" onChange={this.handleEmailTwo.bind(this)} placeholder={this.state.s2email} required/>
+                            </div>
+
+                            <div className="form-child second-row">
+                                <h5 className="label-component" id="info">Email notifications will be sent to all deliverers.</h5>
+                            </div>
+                            <div className="form-child second-row form-buttons-container">
+                                <button type="button" className="form-button" onClick={this.props.handleCancelClick}>Cancel</button>
+                                <button type="submit" className="form-button confirm-button" onSubmit={this.handleConfirmClick.bind(this)}>Confirm</button>
```

Comment:
I think there's an opportunity to refactor here. Instead of having all input fields have a specific handler `onChange`, you can wrap all the inputs under a single parent `<form onSubmit={this.handleConfirm}>...</form>` and save the values in a single function.

Reply:
True, but then targeting all the different values from one event would mean adding different IDs to the input elements. I can do that but I think both of them achieve the same functionality without much difference in logic or code.

Reply:
I agree with Joyce here. The code would be different, you wouldn't need the 6 different functions below that all does pretty much the same thing, and they are actually not necessary. You don't need to update the `state` really when they enter each input box. All we need is at the end, when the `form` is submitted, we grab the inputs and call `handleConfirmClick()` to write the new info to db. 

To change to `<form>`, you also don't need different IDs, you just need the `name` field in the `input`s, which you already have. You just need to make them different.

Reply:
Agreed but don't you think we would need separate onChange handlers for each of the input fields?